tiledb/sm/config/config.h

```diff
@@ -150,6 +150,23 @@ class Config {
    */
   static const std::string SM_CONSOLIDATION_STEP_SIZE_RATIO;
 
+  /**
+   * The consolidation mode. It can be one of:
+   *     - "fragments": only the fragments will be consolidated
+   *     - "fragment_meta": only the fragment metadata will be consolidated
+   *     - "array_meta": only the array metadata will be consolidated
```

Comment:
Also, would this make sense as an enum argument to `tiledb_array_consolidate` in order to be more explicit?

Reply:
Yes we. will deprecate `tiledb_array_consolidate_metadata `. Yeah, an argument makes sense but it would break the API. That's why I thought to add a config parameter instead. I am open to adding the argument.